### DISCUSSION TITLE
using live data

### DISCUSSION THREAD

**Discussion by evieeangeline** [bczsalba/pytermgui#77] [Q&A]
Hi! I have some live data that I'm looking to display - from a subscribe type system so would be a for loop of new data. 

just am a bit confused with how the windows/widgets refresh - is there an example of refreshing data that I could take a look at ? whenever I set up a for loop it gets stuck in the loop instead of refreshing

do I make sense? thanks so much!

e.g. 
```
def main(argv: list[str] | None = None) -> None:
    n=0 

    with ptg.WindowManager() as manager:
        manager.layout = glf._define_layout() # layout settings elsewhere

        # window
        meter_window = ptg.Window("meter_window {n}")
        manager.add(
            ptg.Window(meter_window),
            assign="meter_statcom",
        )
        
        while True: #not my live data but a loop example
            n = n+1
            meter_window = ptg.Window("meter_statcom {n}")
    
```

**Comment by 4l1fe** [comment]
I'm looking for something similar to output logs.

It seems like for now there is no an interface for streaming data yet.
